components/match2/commons/match_group_input.lua

```diff
@@ -241,6 +243,39 @@ function MatchGroupInput.getInexactDate(suggestedDate)
 	return getContentLanguage():formatDate('c', inexactDateString)
 end
 
+--[[
+Parses the match group context. The match group context describes where a match
+group is relative to the tournament page.
+]]
+function MatchGroupInput.readContext(args)
+	return {
+		bracketIndex = tonumber(globalVars:get('match2bracketindex')) or 0,
+		groupRoundIndex = MatchGroupInput.readGroupRoundIndex(args),
+		matchSection = args.matchsection or globalVars:get('matchsection'),
+		sectionHeader = args.section or globalVars:get('bracket_header'),
+		tournamentParent = globalVars:get('tournament_parent'),
+	}
+end
+
+function MatchGroupInput.readGroupRoundIndex(args)
+	if args.round then
+		return tonumber(args.round)
+	end
+
+	local matchSection = args.matchsection or globalVars:get('matchsection')
+	-- Usually 'Round N' but can also be 'Day N', 'Week N', etc.
+	local roundIndex = matchSection and matchSection:match(' (%d+)$')
+	return roundIndex and tonumber(roundIndex)
```

Comment:
How would this handle eg. 
https://liquipedia.net/dota2/The_International/2021/Group_Stage#Matches

Reply:
Each round spans multiple matchlists so the `MatchSection` template can be used. e.g. `{{MatchSection|Day 5}}`